foundation/src/main/java/org/cbio/portal/pipelines/foundation/util/GeneDataUtils.java

```diff
@@ -0,0 +1,243 @@
+                line = buff.readLine();
+            }
+            // extract the maf file header
+            columnNames = splitDataFields(line);
+            reader.close();
+        }
+        
+        return columnNames;
+    }
+    
+    /**
+     * Determines whether given symbol is a ncRNA.
+     * @param hugoSymbol
+     * @return 
+     */
+    public boolean isNcRNA(String hugoSymbol) {
+        return (ncRNABlackList.contains(hugoSymbol) || 
+                Strings.isNullOrEmpty(hugoSymbol) || 
+                FoundationUtils.NULL_EMPTY_VALUES.contains(hugoSymbol));
+    }
+    
+    /**
+     * Resolves list of gene symbols. 
+     * @param geneList
+     * @return 
+     */
+    public List<String> resolveGeneSymbolList(String[] geneList) {
+        List<String> resolvedGeneList = new ArrayList();        
+        if (geneList.length == 1) {
+            // if only one gene in list then resolve single gene symbol
+            resolvedGeneList.add(resolveGeneSymbol(geneList[0]));
+        }
+        else {
+            // if more than one gene in list then it is assumed that the first
+            // gene in list contains the gene name prefix as in the case of 
+            // CDKN2A/B, which has a gene name prefix of 'CDKN2'
+            String genePrefix = geneList[0].substring(0, geneList[0].length()-geneList[1].length());
+            for (int i=0;i<geneList.length;i ++) {
+                String resolvedGene;
+                if (i==0) {
+                    resolvedGene = resolveGeneSymbol(geneList[i]);
+                }
+                else {
+                    resolvedGene = resolveGeneSymbol(genePrefix + geneList[i]);                    
+                }
+                resolvedGeneList.add(resolvedGene);
+            }
+        }
+        return resolvedGeneList;
+    }
+    
+    /**
+     * Resolves the given gene symbol.
+     * Returns the input gene symbol if could not be resolved.
+     * @param hugoSymbol
+     * @return 
+     */
+    public String resolveGeneSymbol(String hugoSymbol) {
+        if (Strings.isNullOrEmpty(hugoSymbol) || FoundationUtils.NULL_EMPTY_VALUES.contains(hugoSymbol)) {
+            return hugoSymbol;
+        }
+        else if (hugoGeneSymbolMap.containsKey(hugoSymbol)) {
+            return hugoSymbol;
+        }
+        else if (geneAliasMap.containsKey(hugoSymbol)) {
+            LOG.info("Gene " + hugoSymbol+ " given as alias - normalizing symbol to: " + geneAliasMap.get(hugoSymbol));
+            return geneAliasMap.get(hugoSymbol);                
+        }
+        else if (hugoSymbol.startsWith("LOC")) {
+            String entrezId = hugoSymbol.replace("LOC", "");
+            if (entrezGeneIdMap.containsKey(entrezId)) {
+                LOG.info("Gene symbol " + hugoSymbol + " contains entrez ID - normalizing symbol to: " + entrezGeneIdMap.get(entrezId));
+                geneAliasMap.put(hugoSymbol, entrezGeneIdMap.get(entrezId));
```

Comment:
Right, sometimes LOC ID#'s are given instead of mapping the ID to a hugo symbol. This scenario might arise if the sequencing platform used doesn't contain updated gene symbol information and post-processing gene symbol mapping doesn't occur